pkg/nanopb/Makefile

```diff
@@ -0,0 +1,30 @@
+PKG_NAME=nanopb
+PKG_URL=https://github.com/nanopb/nanopb.git
+PKG_VERSION=be0dd4b3aacdc9c2bbe21568b6a37218f2bb959a
+PKG_LICENSE=zlib
+
+.PHONY: all
+
+RIOT_CFLAGS = $(INCLUDES) -std=c99
+
+TOOLCHAIN_FILE = $(PKG_BUILDDIR)/xcompile-toolchain.cmake
+
+PKG_DIR=$(CURDIR)
+PROTO_DIR=$(APPDIR)/proto
+PROTO_BUILD_DIR=$(APPDIR)/proto_compiled
+PROTO_FILES=$(shell ls $(PROTO_DIR))
+
+all: $(PKG_BUILDDIR)/Makefile
+	cp $(PKG_BUILDDIR)/libnanopb.a $(BINDIR)/nanopb.a
+
+$(PKG_BUILDDIR)/Makefile: $(TOOLCHAIN_FILE)
+	rm -rf $(PROTO_BUILD_DIR) && \
+	mkdir $(PROTO_BUILD_DIR) && \
+	cp $(PKG_DIR)/Makefile.template $(PKG_BUILDDIR)/Makefile && \
+	$(MAKE) -e PROTO_BUILD_DIR="$(PROTO_BUILD_DIR)" -e PROTO_FILES="$(PROTO_FILES)" \
```

Comment:
The processing of the `.proto` files should not be done by the package's makefile.

Reply:
Why shouldn't it? Otherwise the user has to take care of it or has to keep some code for it in the applications Makefile. I want to keep it as simple as possible for the user.

Reply:
The package's makefile should only take care of building the package. The rest of the rules must be in Makefile.include so that they are added to the main makefile. It is possible to define it so that the user does not have to worry about anything: the same way it is done for `.c` files:

- Use a wildcard to collect `.proto` files
- From this collection, create a target for each .proto.
- Define an implicit rule so that make knows how to build a `pb.{c,h}` from a `.proto`